snuba/utils/streams/batching.py

```diff
@@ -175,6 +175,9 @@ def _handle_message(self, msg: Message[TStream, TOffset, TValue]) -> None:
         if result is not None:
             self.__batch_results.append(result)
 
+        # XXX: abstraction leak/invalid type
+        self.consumer.stage_offsets({msg.stream: msg.offset + 1})
```

Comment:
I kind of expected this to happen earlier, but here it finally is. I see a two ways of dealing with this:

1. Remove `TOffset`, assume that all offsets are natural numbers, use arithmetic to get the offset of the next message.
2. Keep `TOffset`, and expose a `next_offset` attribute on the `Message` object.